ir-test.cxx

```diff
@@ -231,25 +233,79 @@ namespace ir {
 	}
 }
 
+static void print_help(void)
+{
+	std::cout << "Run IR uint tests\n";
+	std::cout << "Usage:\n  ";
+	std::cout << ::this_exe
+	          << " [--show-diff] [--no-color] [test folders or files...]\n";
+	std::cout << "  Run all tests if no test folders/files are specified\n";
+}
 int main(int argc, char **argv) {
+	// save this executable path to infer build_dir and test_dir
+	::this_exe = argv[0];
+	::this_exe_path = std::filesystem::path{::this_exe}.parent_path();
+
+	// store test folders/files specified by user
+	std::set<std::string> user_files;
+	bool opt_error = false;
+
 	for (int i = 1; i < argc; i++) {
 		// XXX use some cleaner arg parsing solution
 		if (!std::string(argv[i]).compare("--show-diff")) {
 			::show_diff = true;
+			continue;
 		} else if (!std::string(argv[i]).compare("--no-color")) {
 			::colorize = false;
+			continue;
+		} else if (!std::string(argv[i]).compare("--help")) {
+			print_help();
+			return 0;
+		} else {
+			// user specified test folders/files
+			std::string file = argv[i];
+			if (std::filesystem::exists(file)
```

Comment:
I would recommend to check for ``--`` prefix and report about unsupported option before trying argument as a file path.

Reply:
will add this check.

Reply:
![image](https://user-images.githubusercontent.com/6504403/232735553-e6e3113b-de7f-4d9d-ae50-949223c1734c.png)